crates/matrix-sdk-common/src/deserialized_responses.rs

```diff
@@ -1247,4 +1248,126 @@ mod tests {
         let reason = UnableToDecryptReason::UnknownMegolmMessageIndex;
         assert!(reason.is_missing_room_key());
     }
+
+    #[test]
+    fn snapshot_test_verification_level() {
+        let level = VerificationLevel::VerificationViolation;
+        insta::assert_json_snapshot! {
+            serde_json::to_value(&level).unwrap(),
+        }
+
+        let level = VerificationLevel::UnsignedDevice;
+        insta::assert_json_snapshot! {
+            serde_json::to_value(&level).unwrap(),
+        }
+
+        let level = VerificationLevel::None(DeviceLinkProblem::InsecureSource);
+        insta::assert_json_snapshot! {
+            serde_json::to_value(&level).unwrap(),
+        }
+
+        let level = VerificationLevel::None(DeviceLinkProblem::MissingDevice);
+        insta::assert_json_snapshot! {
+            serde_json::to_value(&level).unwrap(),
+        }
+
+        let level = VerificationLevel::UnverifiedIdentity;
+        insta::assert_json_snapshot! {
+            serde_json::to_value(&level).unwrap(),
+            @r###""UnverifiedIdentity""###
+        }
+    }
+
+    #[test]
+    fn snapshot_test_verification_states() {
+        let states = vec![
```

Comment:
Is there a way to iterate other all enums? I can see some macro that do that. It might be interesting to do it here?

Reply:
What you are looking for is Strum: https://docs.rs/strum/latest/strum/trait.VariantIterator.html. 

I don't think it's worth adding this trait to the enum, unless we need an iterator over the variants in the public API as well.